### DISCUSSION TITLE
UUID does not set 'pkname'

### DISCUSSION THREAD

**Discussion by yudjinn** [collerek/ormar#510] [Q&A]
I have the following setup: 
models
 ├─base.py 
 ├─user.py
 └─item.py

Base is defined as such:
`
class BaseMeta(ormar.ModelMeta):
    database = database
    metadata = metadata

class Base(ormar.Model, DateFieldsMixins):
    class Meta(BaseMeta):
        abstract = True

    id: UUID4 = ormar.UUID(primary_key=True, default=uuid.uuid4)
`
User: 
`class User(Base):
    class Meta:
        abstract = True
        tablename = "users"

    username: str = ormar.String(max_length=50, unique=True, index=True)
    is_active: bool = ormar.Boolean(default=True)

class UserDB(User):
    hashed_password: str = ormar.String(max_length=256)
`
and Item: 
`
class Item(Base):
    class Meta:
        tablename = "items"

    name: str = ormar.String(max_length=50, unique=True, index=True)
    owner: Optional[Union[User, Dict]] = ormar.ForeignKey(
        UserDB,
        related_name="items",
    )
`

when running `alembic revision --autogenerate -m "init"` I get:
AttributeError: type object 'Meta' has no attribute 'pkname' from ormar/fields/foreign_key.py:114

I'm guessing the UUID column does not set the primary key attribute on the meta class? The docs didnt seem to answer how to handle mutli-file models so I may have missed something.

**Comment by yudjinn** [comment]
Also, do you have to redefine Meta in EVERY class definition? I was hoping to extend the Base class and have the database etc be inherited

**Comment by SepehrBazyar** [comment]
Hi @yudjinn 
This Problem you mentioned has nothing to do with `UUID` and I suggest you read this [best practice](https://collerek.github.io/ormar/models/#best-practice) for `Meta` class and also read [this section](https://collerek.github.io/ormar/models/inheritance/) for document type of inheritance.

Your Code was a bit messy and I couldn't read it well, but if you like it, I have a project in [this link](https://github.com/SepehrBazyar/TaskAPI/tree/test_apis) where you can look at how to define models and inheritance and their relationship with each other. I hope it will be useful and useful to you.
